src/vehicle.cpp

```diff
@@ -634,9 +633,21 @@ void vehicle::control_engines() {
         cruise_velocity = safe_vel;
     }
 
-    //if an engine isn't running, and player is in control, need to start engine.
-    if (g->u.controlling_vehicle && !engine_on) {
-        start_engine();
+    // Restart engines with new configuration
+    if( dirty ) {
```

Comment:
You could flip this condition to lose a level of indentation, and you could combine some of the below conditions I think.

```cpp
if (!dirty) {
    return;
}

if (!engine_on) {
    add_msg( _("You change the %s's engine configuration."), name.c_str() );
    return;
}

add_msg( _("You turn off the %s's engines to change their configurations."), name.c_str() );
if (!g->u.controlling_vehicle) {
    return;
}

if (!start_engines() && !velocity) {
    g->u.controlling_vehicle = false;
    add_msg( _("You let go of the controls.") );
}
```